virttest/utils_net.py

```diff
@@ -3007,26 +3007,28 @@ def get_ip_address_by_interface(ifname):
             "Error while retrieving IP address from interface %s." % ifname)
 
 
-def get_host_ip_address(params):
+def get_host_ip_address(params, ip_version="ipv4"):
     """
     Returns ip address of host specified in host_ip_addr parameter if provided.
     Otherwise ip address on interface specified in netdst parameter is returned.
     In case of "nettype == user" "netdst" parameter is left empty, then the
     default interface of the system is used.
     :param params
+    :param ip_version: Host IP version, ipv4 or ipv6.
     """
     host_ip = params.get('host_ip_addr', None)
     if host_ip:
         logging.debug("Use IP address at config %s=%s", 'host_ip_addr', host_ip)
         return host_ip
     net_dev = params.get('netdst')
-    if not net_dev:
+    if net_dev not in get_host_iface():
```

Comment:
>I prefer simple code like:
default_dev = get_host_default_gateway(iface_name=True)
net_dev = params.get("netdst", default_dev)
return get_ip_address_by_interface(netdev, ip_ver)

ok, I can change it this way

> Why not make get_ip_address_by_interface() support get IPv6 address?

@xutian I do not know how to do this using similar approach as in get_ip_address_by_interface() (e.g. fcntl.ioctl and socket). I would have to divide this function by condition into two branches ipv4 & ipv6 - it does not seem to me practical enough. It seems easier to use get_net_if_addrs. What is your opinion?